sqlglot/schema.py

```diff
@@ -494,16 +497,17 @@ def ensure_column_mapping(mapping: t.Optional[ColumnMapping]) -> t.Dict:
 
 
 def flatten_schema(
-    schema: t.Dict, depth: int, keys: t.Optional[t.List[str]] = None
+    schema: t.Dict, depth: t.Optional[int] = None, keys: t.Optional[t.List[str]] = None
 ) -> t.List[t.List[str]]:
     tables = []
     keys = keys or []
+    depth = dict_depth(schema) - 1 if depth is None else depth
 
     for k, v in schema.items():
-        if depth >= 2:
-            tables.extend(flatten_schema(v, depth - 1, keys + [k]))
-        elif depth == 1:
+        if depth == 1 or not isinstance(v, dict):
```

Comment:
when can `v` _not_ be a `dict`?

Reply:
when the schema has different levels as show in the test

Reply:
I see, makes sense